### ISSUE TITLE
Cmd-Enter doesn't execute code if code chunk is not closed

### ISSUE THREAD

**Issue by wch** [rstudio/rstudio#2454]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.440
    OS Version      : Mac
    R Version       : 3.4.3

### Steps to reproduce the problem

Create a R Markdown document with the following:

    ```{r}
    1+1

Note that there are no closing backticks for the chunk. In this document, moving the cursor to the `1+1` line and pressing Cmd-Enter has no effect.

**Comment by ronblum** [comment]
Reproduced in MacOS, Windows, and Linux.

**Comment by kevinushey** [comment]
This is somewhat intentional. In R Notebooks, if the chunk has no end, where are we supposed to display outputs?

We could perhaps show a warning dialog or something similar, but I imagine that users almost never work in unclosed chunks.

**Comment by wch** [comment]
I agree about the Notebook scenario, but I have mine configured to run code in the console. When I'm writing new stuff at the end of a document, I like testing it out as I write by hitting cmd-enter.

**Comment by kevinushey** [comment]
You can insert chunks using <kbd>Ctrl + Alt + I</kbd>.

There's too much code in RStudio that assumes chunks are well-bounded so I don't think we can attempt to support this particular use case.

**Comment by braunsb** [comment]
Perhaps not—but I just spent 15 minutes reinstalling R & RStudio yesterday because my code wouldn't execute. I thought I needed an upgrade but all I needed was the closing tics for my code chunk. Some visual feedback that I was trying to execute code in an unclosed chunk would have saved me some time. Admittedly, though, it was rather dim of me to forget to pair "Code silently refuses to execute" with "check tic marks" the last time I had this problem. I love RStudio generally so this is not a big complaint, just a thought for improving the user's experience. :)

**Comment by MariaSemple** [comment]
@wch has confirmed this issue is still a problem. Adding some visual indicator that the user is trying to execute a closed code chunk would be useful.
